cmd/spire-server/cli/run/run_posix_test.go

```diff
@@ -267,3 +274,12 @@ func newServerConfigCasesOS() []newServerConfigCase {
 func testParseConfigGoodOS(t *testing.T, c *Config) {
 	assert.Equal(t, c.Server.SocketPath, "/tmp/spire-server/private/api-test.sock")
 }
+
+func getAvailablePort() (string, error) {
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return "", err
+	}
+	defer l.Close()
+	return strconv.Itoa(l.Addr().(*net.TCPAddr).Port), nil
```

Comment:
```suggestion
	addrPort, err := netip.ParseAddrPort(l.Addr().String())
	if err != nil {
		return "", err
	}
	
	return strconv.Itoa(int(addrPort.Port())), nil
```
This is a safer way to parse the port number from the address, because `net.Listener.Addr()` returns an implementation of `net.Addr` that is not guaranteed to be `*net.TCPAddr`.